PowerPointLabs/PowerPointLabs/ActionFramework/TooltipsLab/AssignTooltip/AssignTooltipActionHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+
+using Microsoft.Office.Interop.PowerPoint;
+
+using PowerPointLabs.ActionFramework.Common.Attribute;
+using PowerPointLabs.ActionFramework.Common.Extension;
+using PowerPointLabs.ActionFramework.Common.Interface;
+using PowerPointLabs.Models;
+using PowerPointLabs.TextCollection;
+using PowerPointLabs.TooltipsLab;
+using PowerPointLabs.Utils;
+
+namespace PowerPointLabs.ActionFramework.TooltipsLab
+{
+    [ExportActionRibbonId(TooltipsLabText.AssignTooltipTag)]
+    class AssignTooltipActionHandler : ActionHandler
+    {
+        protected override void ExecuteAction(string ribbonId)
+        {
+            Selection selection = this.GetCurrentSelection();
+            PowerPointSlide currentSlide = this.GetCurrentSlide();
+
+            if (!ShapeUtil.IsSelectionShape(selection))
+            {
+                return;
+            }
+
+            try
+            {
+                AssignTooltip.AddTriggerAnimation(currentSlide, selection);
+                // Open the animation custom pane if it is not opened
+                if (!this.GetApplication().CommandBars.GetPressedMso("AnimationCustom"))
+                {
+                    this.GetApplication().CommandBars.ExecuteMso("AnimationCustom");
+                }
+            }
+            catch (Exception)
+            {
+
```

Comment:
Seems like not very nice to catch a generic exception and then leave this part totally empty... consider passing the error message with the exception and showing a message box with the error message?